sys/vane/clay.hoon

```diff
@@ -788,25 +842,81 @@
+    ::  if we find no change, store request for later.
+    ::  %next is just %mult with one path, so we pretend %next = %mult here.
+        ?($next $mult)
+      |^
+      =+  aey=(case-to-aeon:ze q.p.rav)
+      ::  if the requested case is in the future, we can't know anything yet.
+      ?~  aey  (store ~ ~ ~)
+      =+  old=(read-all-at q.p.rav)
       =+  yon=+((need (case-to-aeon:ze q.p.rav)))
-      |-  ^+  +>.^$
+      |-  ^+  ..start-request
+      ::  if we need future revisions to look for change, wait.
       ?:  (gth yon let.dom)
-        (duce -.rav p.rav u.ver)
-      =+  var=(aver p.rav(q [%ud yon]))
-      ?~  var
-        ~&  [%oh-no rave=rav aeon=yon letdom=let.dom]
-        +>.^$
-      ?~  u.var
-        (blab hen p.rav %& %null [%atom %n ~] ~)          ::  only her %x
-      ?:  (equivalent-data:ze u.u.ver u.u.var)
-        $(yon +(yon))
-      (blab hen p.rav u.u.var)
+        (store `yon old ~)
+      =+  new=(read-all-at [%ud yon])
+      ::  if we don't know everything now, store the request for later.
+      ?.  &((levy ~(tap by old) know) (levy ~(tap by new) know))
+        (store `yon old new)
+      ::  if we do know everything now, compare old and new.
+      ::  if there are differences, send response. if not, try next aeon.
+      =;  res
+        ?~  res  $(yon +(yon))
+        (respond res)
+      %+  roll  ~(tap by old)
+      |=  $:  {pax/path ole/(unit cach)}
+              res/(map mood (each cage lobe))
+          ==
+      =+  neu=(~(got by new) pax)
+      ?<  |(?=($~ ole) ?=($~ neu))
+      =-  ?~(- res (~(put by res) u.-))
+      ^-  (unit (pair mood (each cage lobe)))
+      =+  mod=[p.p.rav [%ud yon] pax]
+      ?~  u.ole
+       ?~  u.neu  ~                                     ::  not added
+       `[mod u.u.neu]                                   ::  added
+      ?~  u.neu
+        `[mod [%& %null [%atom %n ~] ~]]                ::  deleted
+      ?:  (equivalent-data:ze u.u.neu u.u.ole)  ~       ::  unchanged
+      `[mod u.u.neu]                                    ::  changed
+      ::
+      ++  store                                         ::  check again later
+        |=  $:  nex/(unit aeon)
+                old/(map path (unit cach))
+                new/(map path (unit cach))
+            ==
+        ^+  ..start-request
+        ?:  ?=($mult -.rav)
+          (duce -.rav p.rav nex old new)
+        %^  duce  -.rav  p.rav
+        =+  ole=~(tap by old)
+        ?>  (lte (lent ole) 1)
+        ?~  ole  ~
+        q:(snag 0 `(list (pair path (unit cach)))`ole)
+      ::
+      ++  respond                                       ::  send changes
+        |=  res/(map mood (each cage lobe))
+        ^+  ..start-request
+        ?:  ?=($mult -.rav)  (blas hen res)
+        ?>  ?=({* $~ $~} res)
+        (blab hen n.res)
+      ::
+      ++  know  |=({p/path c/(unit cach)} ?=(^ c))      ::  know awout file
+      ::
+      ++  read-all-at                                   ::  initialize cache
```

Comment:
Isn't this more general than just for initializing a cache? It's really more like:
`try to read all paths at case, producing a list of (possibly blocked) results`, or something like that.